src/vse_sync_pp/heatmap.py

```diff
@@ -0,0 +1,53 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+
+"""Heatmap data parsed from log messages from a single source."""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.patches as patches
+
+
+class Heatmap():
+    """Heatmap visualization for state transitions"""
+    def __init__(self, x_ticks, y_ticks, title, unallowed_cells,
+                 colorbar_label, xlabel, ylabel):
+        self._x_ticks = x_ticks
+        self._y_ticks = y_ticks
+        self._title = title
+        self._unallowed_cells = unallowed_cells
+        self._colorbar_label = colorbar_label
+        self._xlabel = xlabel
+        self._ylabel = ylabel
+
+    def plot(self, data, filename):
+        np_data = np.array(data)
+        fig, ax = plt.subplots()
+        im = ax.imshow(np_data, cmap="cividis")
+
+        # Add a colorbar for reference
+        cbar = plt.colorbar(im)
+        cbar.set_label(self._colorbar_label)
+
+        # Show all ticks and label them with the respective list entries
+        ax.set_xticks(np.arange(len(self._x_ticks)), labels=self._x_ticks)
+        ax.set_yticks(np.arange(len(self._y_ticks)), labels=self._y_ticks)
+
+        # Rotate the tick labels and set their alignment.
+        plt.setp(ax.get_xticklabels(), rotation=45, ha="right",
+                 rotation_mode="anchor")
+        plt.xlabel(self._xlabel)
+        plt.ylabel(self._ylabel)
+
+        # Loop over data dimensions and create text annotations.
+        for i in range(len(self._x_ticks)):
+            for j in range(len(self._y_ticks)):
+                ax.text(j, i, np_data[i, j],
+                        ha="center", va="center", color="white")
+
+        for row, col in self._unallowed_cells:
```

Comment:
Recommend we document what the caller should supply in `unallowed_cells` and how its used.